Automattic/vip-go-mu-plugins
# Offload term edits to cron

## Description

Depends on https://github.com/Automattic/ElasticPress/pull/42

There's a limit to how many posts you can queue in-place for reindexing. In light of some term circumstances, term edits should be offloaded to a cronjob that then gathers and queues all the posts that need to be reindexed.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Pull down PR.
2. Pull down https://github.com/Automattic/ElasticPress/pull/42
2. Find the term taxonomy id of a taxonomy with posts.
3. Schedule the cron job `wp cron event schedule 'vip_search_term_update' --term=<term taxonomy id you chose>`
4. Run the cron job `wp cron event run 'vip_search_term_update'`
5. Navigate around the main page to trigger it.
6. Monitor the ` wp_vip_search_index_queue` db table for the post ids of the term taxonomy id you used.
7. All posts from the term taxonomy id should have been queued.
